tap_deputy/client.py

```diff
@@ -30,6 +49,28 @@ def __exit__(self, type, value, traceback):
         self.__session.close()
 
     def refresh(self):
+        """
+        Checks token expiry and refresh token when it is expired and dev mode is not enabled
+        """
+        if self.__dev_mode:
+            try:
+                config = utils.read_config(self.__config_path)
+                self.__access_token = config['access_token']
+                self.__refresh_token = config['refresh_token']
+                self.__expires_at=strptime_to_utc(config['expires_at'])
+            except KeyError as _:
+                LOGGER.fatal("Unable to locate key %s in config",_)
+                raise Exception("Unable to locate key in config")
+
+            if self.__access_token and self.__expires_at < now():
+                LOGGER.fatal("Access Token in config is expired, unable to authenticate in dev mode")
```

Comment:
This log is redundant. The [top level exception handler](https://github.com/singer-io/singer-python/blob/0c066de21111d8572425083b4a8792d193c80af1/singer/utils.py#L231-L232) will log this message for you.

With Line 66's log.fatal included here:
```sh
$ tap-deputy -c config -dev
WARNING Executing Tap in Dev mode
CRITICAL Access Token in config is expired, unable to authenticate in dev mode
CRITICAL Access Token in config is expired, unable to authenticate in dev mode
Traceback (most recent call last):
  File "/usr/local/share/virtualenvs/tap-deputy/bin/tap-deputy", line 33, in <module>
    sys.exit(load_entry_point('tap-deputy', 'console_scripts', 'tap-deputy')())
  File "/home/vagrant/git/singer-python/singer/utils.py", line 235, in wrapped
    return fnc(*args, **kwargs)
  File "/home/vagrant/git/tap-deputy/tap_deputy/__init__.py", line 51, in main
    sync(client,
  File "/home/vagrant/git/tap-deputy/tap_deputy/sync.py", line 88, in sync
    catalog = discover(client)
  File "/home/vagrant/git/tap-deputy/tap_deputy/discover.py", line 120, in discover
    schema_dict, metadata = get_schema(client, resource_name)
  File "/home/vagrant/git/tap-deputy/tap_deputy/discover.py", line 74, in get_schema
    data = client.get(
  File "/home/vagrant/git/tap-deputy/tap_deputy/client.py", line 137, in get
    return self.request('GET', path=path, **kwargs)
  File "/usr/local/share/virtualenvs/tap-deputy/lib/python3.8/site-packages/backoff/_sync.py", line 94, in retry
    ret = target(*args, **kwargs)
  File "/home/vagrant/git/tap-deputy/tap_deputy/client.py", line 106, in request
    self.refresh()
  File "/home/vagrant/git/tap-deputy/tap_deputy/client.py", line 67, in refresh
    raise Exception("Access Token in config is expired, unable to authenticate in dev mode")
Exception: Access Token in config is expired, unable to authenticate in dev mode
```

Without Line 66's `log.fatal`:
```sh
$ tap-deputy -c config -dev
WARNING Executing Tap in Dev mode
CRITICAL Access Token in config is expired, unable to authenticate in dev mode
Traceback (most recent call last):
  File "/usr/local/share/virtualenvs/tap-deputy/bin/tap-deputy", line 33, in <module>
    sys.exit(load_entry_point('tap-deputy', 'console_scripts', 'tap-deputy')())
  File "/home/vagrant/git/singer-python/singer/utils.py", line 235, in wrapped
    return fnc(*args, **kwargs)
  File "/home/vagrant/git/tap-deputy/tap_deputy/__init__.py", line 51, in main
    sync(client,
  File "/home/vagrant/git/tap-deputy/tap_deputy/sync.py", line 88, in sync
    catalog = discover(client)
  File "/home/vagrant/git/tap-deputy/tap_deputy/discover.py", line 120, in discover
    schema_dict, metadata = get_schema(client, resource_name)
  File "/home/vagrant/git/tap-deputy/tap_deputy/discover.py", line 74, in get_schema
    data = client.get(
  File "/home/vagrant/git/tap-deputy/tap_deputy/client.py", line 137, in get
    return self.request('GET', path=path, **kwargs)
  File "/usr/local/share/virtualenvs/tap-deputy/lib/python3.8/site-packages/backoff/_sync.py", line 94, in retry
    ret = target(*args, **kwargs)
  File "/home/vagrant/git/tap-deputy/tap_deputy/client.py", line 106, in request
    self.refresh()
  File "/home/vagrant/git/tap-deputy/tap_deputy/client.py", line 67, in refresh
    raise Exception("Access Token in config is expired, unable to authenticate in dev mode")
Exception: Access Token in config is expired, unable to authenticate in dev mode
```

Reply:
Here's the config I used to produce this

```json
{
  "start_date": "2022-01-01T00:00:00Z",
  "domain": "foo",
  "client_id": "123",
  "client_secret": "456",
  "redirect_uri": "bar",
  "refresh_token": "token",
  "access_token": "token",
  "expires_at": "2022-01-01T00:00:00Z"
}
```